### ISSUE TITLE
Multiple Identical Traces in Annotations

### ISSUE THREAD

**Issue by ev-anderson** [seisbench/seisbench#196]
Hello,

I am working through the model_api example, but instead applying the "obs" model and using some 200 hz OBS data. When I apply the annotate function to my data, I get 6 traces in the resulting Stream object; which are P,S,N,P,S,N--clearly multiples. I've also adjusted the 'model.sampling_rate = 200'.

Is this due to using 200hz data? Or the addition of the H-channel?

Thank you!

**Comment by yetinam** [comment]
Hi @ev-anderson ,
thanks for trying out PickBlue and the issue report. In general, neither the H-channel nor the 200 Hz data should cause a duplication in the output. To help debug the issue, could you maybe share the example data where the issue occurs? If that is not possible, at least the metadata of `stream` and the output of `annotate` would be helpful (as in the example below).

```
In [8]: stream
Out[8]: 
4 Trace(s) in Stream:
IQ.OBS1..HHE | 2016-07-03T22:50:26.990900Z - 2016-07-03T22:51:41.990900Z | 50.0 Hz, 3751 samples
IQ.OBS1..HHH | 2016-07-03T22:50:26.990900Z - 2016-07-03T22:51:41.990900Z | 50.0 Hz, 3751 samples
IQ.OBS1..HHN | 2016-07-03T22:50:26.990900Z - 2016-07-03T22:51:41.990900Z | 50.0 Hz, 3751 samples
IQ.OBS1..HHZ | 2016-07-03T22:50:26.990900Z - 2016-07-03T22:51:41.990900Z | 50.0 Hz, 3751 samples

In [9]: model.annotate(stream)
Out[9]: 
3 Trace(s) in Stream:
IQ.OBS1..PhaseNet_P | 2016-07-03T22:50:26.990900Z - 2016-07-03T22:51:42.000900Z | 100.0 Hz, 7502 samples
IQ.OBS1..PhaseNet_S | 2016-07-03T22:50:26.990900Z - 2016-07-03T22:51:42.000900Z | 100.0 Hz, 7502 samples
IQ.OBS1..PhaseNet_N | 2016-07-03T22:50:26.990900Z - 2016-07-03T22:51:42.000900Z | 100.0 Hz, 7502 samples

```

**Comment by ev-anderson** [comment]
![Screen Shot 2023-05-01 at 12 10 42 PM](https://user-images.githubusercontent.com/89091882/235513216-14b5058a-9488-41ad-baa6-14db3061bd20.png)

**Comment by ev-anderson** [comment]
![Screen Shot 2023-05-01 at 12 10 42 PM](https://user-images.githubusercontent.com/89091882/235513522-3e66ff67-156e-4930-a9c3-f13502cf3a08.png)

**Comment by yetinam** [comment]
Thanks for the quick feedback. The issue is that SeisBench uses the everything except the component code to distinguish between instruments, i.e., in your case `X3.OBS60..ED` and `X3.OBS60..EL`. This mean, it finds to instruments. However, when naming the output channels, the full channel code (last three letters) is truncated, so the output appears duplicate. As a workaround, just rename the channel for the hydrophone component to `ELH`.

I'll think about a more long-term solution.

**Comment by yetinam** [comment]
The issue is now fixed in the `main` branch. The fix will be included in v0.5.
